openedx/core/djangoapps/credit/tasks.py

```diff
@@ -33,23 +33,94 @@ def update_course_requirements(course_id):
         is_credit_course = CreditCourse.is_credit_course(course_key)
         if is_credit_course:
             course = modulestore().get_course(course_key)
-            requirements = [
-                {
-                    "namespace": "grade",
-                    "name": "grade",
-                    "criteria": {
-                        "min_grade": get_min_grade_for_credit(course)
-                    }
-                }
-            ]
+            requirements = _get_course_credit_requirements(course)
             set_credit_requirements(course_key, requirements)
-    except (InvalidKeyError, ItemNotFoundError, InvalidCreditRequirements) as exc:
+    except (InvalidKeyError, ItemNotFoundError, InvalidCreditRequirements, AttributeError) as exc:
         LOGGER.error('Error on adding the requirements for course %s - %s', course_id, unicode(exc))
         raise update_course_requirements.retry(args=[course_id], exc=exc)
     else:
         LOGGER.info('Requirements added for course %s', course_id)
 
 
-def get_min_grade_for_credit(course):
-    """ Returns the min_grade for the credit requirements """
-    return getattr(course, "min_grade", 0.8)
+def _get_course_credit_requirements(course):
+    """ Returns the list of credit requirements for the given course
+
+    It returns the minimum_grade_credit and also the ICRV checkpoints
+    if added any in the course
+
+    Args:
+        course(Course): The course object
+
+    Returns:
+        List of minimum_grade_credit and ICRV requirements
+    """
+    icrv_requirements = _get_credit_course_requirement_xblocks(course)
+    min_grade_requirement = _get_min_grade_requirement(course)
+    icrv_requirements.extend(min_grade_requirement)
+    return icrv_requirements
+
+
+def _get_min_grade_requirement(course):
+    """ Returns the list of minimum_grade_credit requirements for the given course
+
+    Args:
+        course(Course): The course object
+
+    Raises:
+        AttributeError if the course has not minimum_grade_credit attribute
+
+    Returns:
+        The list of minimum_grade_credit requirements
+    """
+    requirement = [
+        {
+            "namespace": "grade",
+            "name": "grade",
+            "criteria": {
+                "min_grade": getattr(course, "minimum_grade_credit")
+            }
+        }
+    ]
+    return requirement
+
+
+def _get_credit_course_requirement_xblocks(course):  # pylint: disable=invalid-name
+    """ Generates a course structure dictionary for the specified course.
```

Comment:
@cpennington We're trying to find XBlocks that satisfy a certain interface.  My thought is that since we're doing this asynchonously, so we're not as concerned about the performance hit of instantiating XBlocks, as long as we don't try to load them all into memory at once.

Does this approach make sense, or are there better ways to do this?  For example, are there query patterns already supported by modulestore that we can take advantage of?